File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<int> stack = new Stack<int>();
        int n = ingredient.Length;
        
        for(int i = 0; i < n; i++)
        {
            stack.Push(ingredient[i]);
            if(stack.Count >= 4)
            {
                int a = stack.Pop();
                int b = stack.Pop();
                int c = stack.Pop();
                int d = stack.Pop();
                if(d == 1 && c == 2 && b == 3 && a == 1)
                {
                    answer++;
                }
                else{
                    stack.Push(d);
                    stack.Push(c);
                    stack.Push(b);
                    stack.Push(a);
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cpp
#include <string>
#include <vector>

using namespace std;
// 빵 - 야채 - 고기 - 빵 -> 1-2-3-1 스택?
int solution(vector<int> ingredient) {
    int answer = 0;
    vector<int> v;
    
    for(int x : ingredient)
    {
        v.push_back(x);
        
        if(v.size() >= 4)
        {
            int n = v.size();
            if(v[n - 1] == 1 && v[n - 2] == 3 && v[n - 3] == 2 && v[n- 4] == 1)
            {
                for(int i = 0; i < 4; i++)
                    v.pop_back();
                
                answer++;
            }
        }
    }
    return answer;
}